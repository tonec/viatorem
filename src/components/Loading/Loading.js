import React from 'react'
import PropTypes from 'prop-types'

const propTypes = {
  isLoading: PropTypes.bool,
  pastDelay: PropTypes.bool,
  error: PropTypes.bool
}

const defaultProps = {
  isLoading: false,
  pastDelay: false,
  error: false
}

const Loading = ({ isLoading, pastDelay, error }) => {
  if (isLoading && pastDelay) {
    return <p>Loading...</p>
  } else if (error && !isLoading) {
    return <p>Error!</p>
  } else {
    return null
  }
}

Loading.propTypes = propTypes
Loading.defaultProps = defaultProps

export default Loading
