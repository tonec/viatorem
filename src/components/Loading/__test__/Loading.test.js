import React from 'react'
import { shallow } from 'enzyme'
import Loading from '../Loading'

describe('Loading (snapshot)', () => {
  it('renders according to snapshot when isLoading & pastDelay are both true', () => {
    const component = shallow(<Loading isLoading pastDelay />)
    expect(component).toMatchSnapshot()
  })

  it('renders according to snapshot when error is true', () => {
    const component = shallow(<Loading error />)
    expect(component).toMatchSnapshot()
  })

  it('renders according to snapshot when no props are passed in', () => {
    const component = shallow(<Loading />)
    expect(component).toMatchSnapshot()
  })
})
