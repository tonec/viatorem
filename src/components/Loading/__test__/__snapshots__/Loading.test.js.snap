// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading (snapshot) renders according to snapshot when error is true 1`] = `
<p>
  Error!
</p>
`;

exports[`Loading (snapshot) renders according to snapshot when isLoading & pastDelay are both true 1`] = `
<p>
  Loading...
</p>
`;

exports[`Loading (snapshot) renders according to snapshot when no props are passed in 1`] = `""`;
